paddle/fluid/operators/distributed/trainer_heart_beat_monitor.h

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <gflags/gflags.h>
+
+#include <functional>
+#include <future>  // NOLINT
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include <thread>  // NOLINT
+
+#include <ThreadPool.h>
+
+#include "paddle/fluid/platform/enforce.h"
+
+namespace paddle {
+namespace operators {
+namespace distributed {
+
+enum TrainerStatus { UNINITED = 0, RUNNING, COMPLETED };
+
+struct Trainer {
+  int id;
+  TrainerStatus status;
+  int timestamp;
+
+  Trainer() {}
+
+  explicit Trainer(int trainer_id) {
+    this->id = trainer_id;
+    this->status = UNINITED;
+    this->timestamp = 0;
+  }
+};
+
+class TrainerHeartBeatMonitor {
```

Comment:
I'm not sure whether this heartBeatMonitor can only be used for Trainer. In general any endpoint can have a HeartBeatMonitor.